### PR TITLE
rename AddCleanup to Defer

### PR DESCRIPTION
**Pull request by rogpeppe**
This naming more succinct, more Go-idiomatic, and more suggestive of the
semantics that are implemented (a LIFO stack of functions to run later).

We also rename Cleanup to Done (thinking of sync.WaitGroup.Done).

The old names are left in place as synonyms for now.

### CODE REVIEW

File: quicktest.go [hunk 1]
```diff
@@ -33,34 +44,45 @@ func New(t testing.TB) *C {
...
-// only call the cleanup functions once.
+// Done calls all the functions registered by Defer in reverse
+// registration order. After it's called, the functions are
+// unregistered, so calling Done twice will only call them once.
 //
 // When a test function is called by Run, the C value passed into it
-// will be cleaned up automatically when it returns.
-func (c *C) Cleanup() {
-	// Note: we need to use defer in case the cleanup panics
+// Done will be called automatically when it returns.
```
Comment (frankban) [inline quicktest.go]: "... the C value passed into it Done will be called ..." Maybe rephrase this?

Comment (rogpeppe) [reply quicktest.go]: done
